server/routes/induction/common/qualificationsListController.ts

```diff
@@ -0,0 +1,66 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import type { AchievedQualificationDto } from 'inductionDto'
+import type { Assessment } from 'viewModels'
+import InductionController from './inductionController'
+import QualificationsListView from './qualificationsListView'
+import dateComparator from '../../dateComparator'
+
+/**
+ * Abstract controller class defining functionality common to both the Create and Update Induction journeys.
+ */
+export default abstract class QualificationsListController extends InductionController {
+  /**
+   * Returns the Qualifications List view; suitable for use by the Create and Update journeys.
+   */
+  getQualificationsListView: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonerSummary, inductionDto, prisonerFunctionalSkills } = req.session
+
+    const qualifications: Array<AchievedQualificationDto> = inductionDto.previousQualifications?.qualifications
+
+    const functionalSkills = {
+      ...prisonerFunctionalSkills,
+      assessments: mostRecentAssessments(prisonerFunctionalSkills.assessments),
+    }
+
+    const view = new QualificationsListView(
+      prisonerSummary,
+      this.getBackLinkUrl(req),
+      this.getBackLinkAriaText(req),
+      qualifications,
+      functionalSkills,
+    )
+    return res.render('pages/induction/prePrisonEducation/qualificationsList', { ...view.renderArgs })
+  }
+}
+
+const mostRecentAssessments = (allAssessments: Array<Assessment>): Array<Assessment> => {
+  const allAssessmentsGroupedByTypeSortedByDateDesc = assessmentsGroupedByTypeSortedByDateDesc(allAssessments)
+
+  const latestEnglishAssessment = allAssessmentsGroupedByTypeSortedByDateDesc.get('ENGLISH')?.at(0)
+  const latestMathsAssessment = allAssessmentsGroupedByTypeSortedByDateDesc.get('MATHS')?.at(0)
+  const latestOtherAssessments = [...allAssessmentsGroupedByTypeSortedByDateDesc.keys()]
+    .filter(key => key !== 'ENGLISH' && key !== 'MATHS')
+    .map(key => allAssessmentsGroupedByTypeSortedByDateDesc.get(key).at(0))
+
+  return Array.of(latestEnglishAssessment, latestMathsAssessment, ...latestOtherAssessments).filter(
+    assessment => assessment != null,
+  )
+}
+
+const assessmentsGroupedByTypeSortedByDateDesc = (assessments: Array<Assessment>): Map<string, Array<Assessment>> => {
+  const map = new Map<string, Array<Assessment>>()
+  assessments.forEach(assessment => {
+    const key = assessment.type
+    const value: Array<Assessment> = map.get(key) || []
+    value.push(assessment)
+    map.set(
+      key,
+      value.sort((left: Assessment, right: Assessment) => dateComparator(left.assessmentDate, right.assessmentDate)),
```

Comment:
Minor - I think it would be good to provide the `DESC` argument here, just to make it clear. As it stands, I had to look at the `dateCompartor` class to see if/how items are coming back in descending order. As an aside, I'm not massively keen on `DESC` being the default, as I think that's counter-intuitive, but that's a separate point!

Reply:
No probs, will do (add the argument as suggested; not change the default behaviour)